accelio: fix i686-linux build

dfoxfranke opened the pull request:
- Compile with gcc5 to avoid the compiler bug described in https://gcc.gnu.org/ml/gcc-patches/2014-05/msg02560.html
- Add a patch to fix the many incorrect printf format specifiers and other sloppy type conversions that gcc5 catches and warns on (erroring out due to -Werror).

cc @wkennington

dfoxfranke committed:
accelio: fix i686-linux build

* Compile with gcc5 to avoid the compiler bug described in
  https://gcc.gnu.org/ml/gcc-patches/2014-05/msg02560.html

* Add a patch to fix the many incorrect printf format specifiers and
  other sloppy type conversions that gcc5 catches and warns on
  (erroring out due to -Werror).

wkennington commented:
Is this patch not included upstream? I'm building accelio fine on gcc5 so
I'm not sure why all of this is needed.

On Sun, Aug 30, 2015, 20:55 Daniel Fox Franke <notifications@github.com>
wrote:

>
>    -
>
>    Compile with gcc5 to avoid the compiler bug described in
>    https://gcc.gnu.org/ml/gcc-patches/2014-05/msg02560.html
>    -
>
>    Add a patch to fix the many incorrect printf format specifiers and
>    other sloppy type conversions that gcc5 catches and warns on (erroring out
>    due to -Werror).
>
> cc @wkennington <https://github.com/wkennington>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/NixOS/nixpkgs/pull/9570
> Commit Summary
>
>    - accelio: fix i686-linux build
>
> File Changes
>
>    - *M* pkgs/development/libraries/accelio/default.nix
>    <https://github.com/NixOS/nixpkgs/pull/9570/files#diff-0> (2)
>    - *A* pkgs/development/libraries/accelio/fix-printfs.patch
>    <https://github.com/NixOS/nixpkgs/pull/9570/files#diff-1> (599)
>    - *M* pkgs/top-level/all-packages.nix
>    <https://github.com/NixOS/nixpkgs/pull/9570/files#diff-2> (2)
>
> Patch Links:
>
>    - https://github.com/NixOS/nixpkgs/pull/9570.patch
>    - https://github.com/NixOS/nixpkgs/pull/9570.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/9570>.
>

dfoxfranke commented:
It errors out for me. Are you turning off `-Werror` or some such?

...and I just noticed it still does on `x86_64-linux` because there's a conversion I missed, so hold off on merging this until I fix that.

wkennington commented:
Nope, but I'm using gcc5 in the stdenv so maybe that changes things?

On Sun, Aug 30, 2015, 21:00 Daniel Fox Franke <notifications@github.com>
wrote:

> It errors out for me. Are you turning off -Werror or some such?
>
> ...and I just noticed it still does on x86_64-linux because there's a
> conversion I missed, so hold off on merging this until I fix that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/9570#issuecomment-136246082>.
>

dfoxfranke commented:
Most of those things I changed from `%lu` to `%llu` actually need to be changed to `PRIu64` (a macro from `<inttypes.h>`) to be portable. I'll fix it tomorrow.

wkennington commented:
You should try and get this upstreamed if not already.

On Sun, Aug 30, 2015, 21:48 Daniel Fox Franke <notifications@github.com>
wrote:

> Most of those things I changed from %lu to %llu actually need to be
> changed to PRIu64 (a macro from <inttypes.h>) to be portable. I'll fix it
> tomorrow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/9570#issuecomment-136250979>.
>

dfoxfranke commented:
I will, once I get it right.
On Aug 31, 2015 12:51 AM, "William A. Kennington III" <
notifications@github.com> wrote:

> You should try and get this upstreamed if not already.
>
> On Sun, Aug 30, 2015, 21:48 Daniel Fox Franke <notifications@github.com>
> wrote:
>
> > Most of those things I changed from %lu to %llu actually need to be
> > changed to PRIu64 (a macro from <inttypes.h>) to be portable. I'll fix it
> > tomorrow.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/NixOS/nixpkgs/pull/9570#issuecomment-136250979>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/9570#issuecomment-136251441>.
>

lethalman commented:
Beware that `%llu` may break the x64 version. Usually the correct format is `%zu`.

dfoxfranke commented:
@lethalman The offending arguments are (mostly) `uint64_t`, not `size_t`.

garbas commented:
@dfoxfranke this PR fails for me on x64

dfoxfranke commented:
@garbas yes, this is currently WIP until I make the `%llu` -> `PRIu64` fix I described above.

domenkozar commented:
cool, let me know when it's ready

dfoxfranke committed:
accelio: enable tests

The patch committed with 88471b684e6544da7691937a9b68cefa49d260d5
makes them work again.

dfoxfranke commented:
I've revised the patch and this should now be ready to go. The patch also fixes all the tests and examples, so in a separate commit I've re-enabled building them and set `doCheck = true`.

I'll submit these changes upstream once I get back from vacation next week.

dfoxfranke commented:
Bah... hold off one more moment. I just want to make a cosmetic change to the patch file.

dfoxfranke commented:
Done.

domenkozar commented:
it breaks kernel module(s) http://hydra.nixos.org/eval/1219660#tabs-now-fail

dfoxfranke commented:
@domenkozar Revert 8b663509b1be9d6e55316e60aef838a352d87647 for the time being.
